doc/user/metrics/README.md

```diff
@@ -13,44 +13,120 @@ By default, the metrics are served on `0.0.0.0:8383/metrics`. To modify the port
         "github.com/operator-framework/operator-sdk/pkg/metrics"
-        "sigs.k8s.io/controller-runtime/pkg/manager"
-        "k8s.io/api/core/v1"
+        "github.com/operator-framework/operator-sdk/pkg/restmapper"
+        sdkVersion "github.com/operator-framework/operator-sdk/version"
+        "github.com/spf13/pflag"
+        v1 "k8s.io/api/core/v1"
         "k8s.io/apimachinery/pkg/util/intstr"
+        "sigs.k8s.io/controller-runtime/pkg/client/config"
+        "sigs.k8s.io/controller-runtime/pkg/manager"
+        logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+        "sigs.k8s.io/controller-runtime/pkg/runtime/signals"
+    )
+
+    // Change below variables to serve metrics on different host or port.
+    var (
+        metricsHost               = "0.0.0.0"
+        metricsPort         int32 = 8383
+        operatorMetricsPort int32 = 8686
     )
+    var log = logf.Log.WithName("cmd")
+    
+    func printVersion() {
+        log.Info(fmt.Sprintf("Go Version: %s", runtime.Version()))
+        log.Info(fmt.Sprintf("Go OS/Arch: %s/%s", runtime.GOOS, runtime.GOARCH))
+        log.Info(fmt.Sprintf("Version of operator-sdk: %v", sdkVersion.Version))
+    }
 
     func main() {
 
         ...
 
-        // Change the below variables to serve metrics on different host or port.
-        var metricsHost = "0.0.0.0"
-        var metricsPort int32 = 8383
-
-        // Pass metrics address to controller-runtime manager
-        mgr, err := manager.New(cfg, manager.Options{
-            Namespace:          namespace,
-            MetricsBindAddress: fmt.Sprintf("%s:%d", metricsHost, metricsPort),
-        })
+       // Pass metrics address to controller-runtime manager
+       // Create a new Cmd to provide shared dependencies and start components
+       mgr, err := manager.New(cfg, manager.Options{
+          Namespace: namespace,
+          MapperProvider:     restmapper.NewDynamicRESTMapper,
+          MetricsBindAddress: fmt.Sprintf("%s:%d", metricsHost, metricsPort),
+       })
 
         ...
 
         // Add to the below struct any other metrics ports you want to expose.
-	    servicePorts := []v1.ServicePort{
-		    {Port: metricsPort, Name: metrics.OperatorPortName, Protocol: v1.ProtocolTCP, TargetPort: intstr.IntOrString{Type: intstr.Int, IntVal: metricsPort}},
-	    }
+      	servicePorts := []v1.ServicePort{
+      		{Port: metricsPort, Name: metrics.OperatorPortName, Protocol: v1.ProtocolTCP, TargetPort: intstr.IntOrString{Type: intstr.Int, IntVal: metricsPort}},
+      		{Port: operatorMetricsPort, Name: metrics.CRPortName, Protocol: v1.ProtocolTCP, TargetPort: intstr.IntOrString{Type: intstr.Int, IntVal: operatorMetricsPort}},
+      	}
+      	// Create Service object to expose the metrics port(s).
+      	service, err := metrics.CreateMetricsService(ctx, cfg, servicePorts)
+      	if err != nil {
+      		log.Info("Could not create metrics Service", "error", err.Error())
```

Comment:
Probably sufficient to continue using `// handle error` here.

We mention after the code block that the runnable metrics code is already present in the default `main.go` scaffold. Perhaps we could make that note more prominent and add a link to the scaffold so that users could see a full example without having to create a new project?